compiler/rustc_hir_analysis/src/hir_wf_check.rs

```diff
@@ -163,15 +163,17 @@ fn diagnostic_hir_wf_check<'tcx>(
                 kind: hir::GenericParamKind::Type { default: Some(ty), .. },
                 ..
             }) => vec![*ty],
-            hir::Node::AnonConst(_)
-                if let Some(const_param_id) =
-                    tcx.hir().opt_const_param_default_param_def_id(hir_id)
+            hir::Node::AnonConst(_) => {
```

Comment:
This needs an explanation for why we're calling `diagnostic_hir_wf_check` on an anon const that has no default.

Reply:
Explanation added above